java/vector/src/main/codegen/templates/UnionLargeListWriter.java

```diff
@@ -0,0 +1,232 @@
+  private long index = 0L;
+  private boolean inStruct = false;
+  private String structName;
+  private static final int OFFSET_WIDTH = 8;
+
+  public UnionLargeListWriter(LargeListVector vector) {
+    this(vector, NullableStructWriterFactory.getNullableStructWriterFactoryInstance());
+  }
+
+  public UnionLargeListWriter(LargeListVector vector, NullableStructWriterFactory nullableStructWriterFactory) {
+    this.vector = vector;
+    this.writer = new PromotableWriter(vector.getDataVector(), vector, nullableStructWriterFactory);
+  }
+
+  public UnionLargeListWriter(LargeListVector vector, AbstractFieldWriter parent) {
+    this(vector);
+  }
+
+  @Override
+  public void allocate() {
+    vector.allocateNew();
+  }
+
+  @Override
+  public void clear() {
+    vector.clear();
+  }
+
+  @Override
+  public Field getField() {
+    return vector.getField();
+  }
+
+  public void setValueCount(int count) {
+    vector.setValueCount(count);
+  }
+
+  @Override
+  public int getValueCapacity() {
+    return vector.getValueCapacity();
+  }
+
+  @Override
+  public void close() throws Exception {
+    vector.close();
+    writer.close();
+  }
+
+  @Override
+  public void setPosition(int index) {
+    throw new UnsupportedOperationException("cant set int index for long list");
+  }
+
+  public void setIndex(long index) {
+    this.index = index;
+  }
+  <#list vv.types as type><#list type.minor as minor><#assign name = minor.class?cap_first />
+  <#assign fields = minor.fields!type.fields />
+  <#assign uncappedName = name?uncap_first/>
+  <#if uncappedName == "int" ><#assign uncappedName = "integer" /></#if>
+  <#if !minor.typeParams?? >
+
+  @Override
+  public ${name}Writer ${uncappedName}() {
+    return this;
+  }
+
+  @Override
+  public ${name}Writer ${uncappedName}(String name) {
+    structName = name;
+    return writer.${uncappedName}(name);
+  }
+  </#if>
+  </#list></#list>
+
+  @Override
+  public DecimalWriter decimal() {
+    return this;
+  }
+
+  @Override
+  public DecimalWriter decimal(String name, int scale, int precision) {
+    return writer.decimal(name, scale, precision);
+  }
+
+  @Override
+  public DecimalWriter decimal(String name) {
+    return writer.decimal(name);
+  }
+
+  @Override
+  public StructWriter struct() {
+    inStruct = true;
+    return this;
+  }
+
+  @Override
+  public ListWriter list() {
+    return writer;
+  }
+
+  @Override
+  public ListWriter list(String name) {
+    ListWriter listWriter = writer.list(name);
+    return listWriter;
+  }
+
+  @Override
+  public StructWriter struct(String name) {
+    StructWriter structWriter = writer.struct(name);
+    return structWriter;
+  }
+
+  @Override
+  public void startList() {
+    vector.startNewValue(index);
+    writer.setPosition(checkedCastToInt(vector.getOffsetBuffer().getLong((index + 1) * OFFSET_WIDTH)));
+  }
+
+  @Override
+  public void endList() {
+    vector.getOffsetBuffer().setLong((index + 1) * OFFSET_WIDTH, writer.idx());
```

Comment:
Here we need to cast the first argument, too.

Reply:
done